pkg/util/ocm/validator.go

```diff
@@ -0,0 +1,45 @@
+package ocm
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"encoding/json"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+)
+
+var ValidOCMResourceTypes = []string{"syncsets", "secrets", "machinepools", "syncidentityproviders"}
+
+func ValidateOCMResourceType(resourceType string) bool {
```

Comment:
I think https://github.com/Azure/ARO-RP/pull/2446 will give us good coverage for this case. Do you think it's still needed?

Reply:
Yes that validation is great. I can update this PR to remove this extra step.

Reply:
@cadenmarchese I removed this extra validation of the `ocmResourceType`